service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -577,21 +580,22 @@ public PaymentSystemResponse saveFullOrderToDB(OrderResponseDto dto, String uuid
 
         getOrder(dto, currentUser, bagsOrdered, sumToPayInCoins, order, orderCertificates, userData);
         eventService.save(OrderHistory.ORDER_FORMED, OrderHistory.CLIENT, order);
+        PaymentSystemResponse paymentSystemResponse = processPayment(dto, order, sumToPayInCoins, currentUser);
 
-        checkIfOrderIsNotPayedAndSendEmailAsync(order, sumToPayInCoins);
+        checkIfOrderIsNotPayedAndSendEmailAsync(order, sumToPayInCoins, paymentSystemResponse);
 
         notificationService.notifyCreatedOrder(order);
 
         if (sumToPayInCoins <= 0 || !dto.isShouldBePaid()) {
             return getPaymentRequestDto(order, "");
         }
-
-        return processPayment(dto, order, sumToPayInCoins, currentUser);
+        return paymentSystemResponse;
     }
 
     @Async
```

Comment:
Directly invoking a method annotated with [@Async or @Transactional](https://www.baeldung.com/spring-transactional-async-annotation) from within the same class
bypasses Spring's proxy mechanism, leading to unexpected behavior (e.g., @Async runs synchronously, 
and @Transactional does not manage transactions). Remove this.

Reply:
I will fix it